src/OrchardCore.Modules/OrchardCore.Autoroute/Services/AutorouteEntries.cs

```diff
@@ -1,92 +1,148 @@
 using System;
 using System.Collections.Generic;
 using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+using OrchardCore.ContentManagement.Records;
 using OrchardCore.ContentManagement.Routing;
+using OrchardCore.Documents;
+using OrchardCore.Environment.Shell.Scope;
+using YesSql;
 
 namespace OrchardCore.Autoroute.Services
 {
     public class AutorouteEntries : IAutorouteEntries
     {
-        private readonly Dictionary<string, AutorouteEntry> _paths;
-        private readonly Dictionary<string, AutorouteEntry> _contentItemIds;
-
         public AutorouteEntries()
         {
-            _paths = new Dictionary<string, AutorouteEntry>();
-            _contentItemIds = new Dictionary<string, AutorouteEntry>(StringComparer.OrdinalIgnoreCase);
         }
 
+        public async Task<(bool, AutorouteEntry)> TryGetEntryByPathAsync(string path)
+        {
+            var document = await GetDocumentAsync();
+
+            if (document.ContentItemIds.TryGetValue(path, out var entry))
+            {
+                return (true, entry);
+            }
+
+            return (false, entry);
+        }
+
+        public async Task<(bool, AutorouteEntry)> TryGetEntryByContentItemIdAsync(string contentItemId)
+        {
+            var document = await GetDocumentAsync();
+
+            if (document.Paths.TryGetValue(contentItemId, out var entry))
+            {
+                return (true, entry);
+            }
 
-        public bool TryGetEntryByPath(string path, out AutorouteEntry entry)
+            return (false, entry);
+        }
+
+        public async Task AddEntriesAsync(IEnumerable<AutorouteEntry> entries)
         {
-            return _contentItemIds.TryGetValue(path, out entry);
+            var document = await LoadDocumentAsync();
+            AddEntries(document, entries);
+            await DocumentManager.UpdateAsync(document);
         }
 
-        public bool TryGetEntryByContentItemId(string contentItemId, out AutorouteEntry entry)
+        public async Task RemoveEntriesAsync(IEnumerable<AutorouteEntry> entries)
         {
-            return _paths.TryGetValue(contentItemId, out entry);
+            var document = await LoadDocumentAsync();
+            RemoveEntries(document, entries);
+            await DocumentManager.UpdateAsync(document);
         }
 
-        public void AddEntries(IEnumerable<AutorouteEntry> entries)
+        // For unit tests only
+        public bool TryGetEntryByPath(AutorouteDocument document, string path, out AutorouteEntry entry)
+            => document.ContentItemIds.TryGetValue(path, out entry);
+
+        // For unit tests only
+        public bool TryGetEntryByContentItemId(AutorouteDocument document, string contentItemId, out AutorouteEntry entry)
```

Comment:
This is annoying just to need these methods for the unit tests

Reply:
Now these methods **use the related document manager**, so for unit tests we would need an ambient shell scope with the related services registered and so on.

So, this is just because i was too lazy to create stub / mock services ;) and i created these methods just to check the inner work as it was done before.

But okay i will update unit tests to use e.g. `TryGetEntryByContentItemIdAsync()` directly